src/classes/UTIL_Describe.cls

```diff
@@ -303,6 +303,109 @@ public class UTIL_Describe {
+
+        return String.isNotBlank(apiName) && apiName.equalsIgnoreCase(fieldName) 
+            ? apiName 
+            : null;
+    }
+    
+    /*********************************************************************************************************
+    * @description Returns API Name for a field or null if the field is invalid 
+    * @param fieldName Field name
+    * @param fieldResultsByFieldName The Sobject's field results mapped by a field name or a relationship name
+    * @return String
+    */
+    private static String getApiName(String fieldName, Map<String, Schema.DescribeFieldResult> fieldResultsByFieldName) {
+        if (String.isBlank(fieldName)) {
+            return null;
+        }
+
+        Schema.DescribeFieldResult fieldResult = fieldResultsByFieldName.get(fieldName.toLowerCase());
+
+        if (fieldResult != null) {
+            return fieldResult.getName();
+        }
+
+        String[] nameParts = fieldName.split('\\.');
+        Boolean isReferenceField = nameParts.size() > 1;
+
+        if (!isReferenceField) { 
+            return null;
+        }
+        
+        String relationshipName = nameParts[0];
+        String referenceFieldName = fieldName.substringAfter('.');
+        Schema.DescribeFieldResult referenceFieldResult = fieldResultsByFieldName.get(relationshipName.toLowerCase());
+
+        return getApiNameOnReferenceSobject(referenceFieldName, referenceFieldResult);
+    }
+
+    /*********************************************************************************************************
+    * @description Returns API Name for a field or null if the field is invalid for the referenced Sobject 
+    * @param fieldName Field name
+    * @param referenceFieldResult The reference (Lookup, M-D) field result
+    * @return String
+    */
+    private static String getApiNameOnReferenceSobject(String fieldName, Schema.DescribeFieldResult referenceFieldResult) {
+        Schema.SObjectType[] references = referenceFieldResult == null 
+            ? null 
+            : referenceFieldResult.getReferenceTo();
+
+        if (String.isBlank(fieldName) || references == null) { 
+            return null; 
+        }
+
+        for (Schema.SObjectType referenceTo : references) {
+            String apiName = getApiName(fieldName, getFieldResultsByName(referenceTo));
+
+            if (String.isNotBlank(apiName)) {
+                return referenceFieldResult.getRelationshipName() + '.' + apiName;
+            }
+        }
+
+        return null;
+    }
+
+    /*********************************************************************************************************
+    * @description Returns Sobject field results mapped by a lower case field name or a relationship name
+    * @param sobjectType Schema.SobjectType
+    * @return Map<String, Schema.DescribeFieldResult>
+    */
+    public static Map<String, Schema.DescribeFieldResult> getFieldResultsByName(Schema.SobjectType sobjectType) {
```

Comment:
New `getFieldResultsByName()` and `getApiName()` methods have been removed from `UTIL_Describe` since they are not used for the Household Naming Settings validation any longer.